playstore/validator.go

```diff
@@ -44,14 +32,24 @@ type Client struct {
 // New returns http client which includes the credentials to access androidpublisher API.
 // You should create a service account for your project at
 // https://console.developers.google.com and download a JSON key file to set this argument.
-func New(jsonKey []byte) (Client, error) {
-	ctx := context.WithValue(oauth2.NoContext, oauth2.HTTPClient, &http.Client{
-		Timeout: timeout,
-	})
+func New(jsonKey []byte) (*Client, error) {
+	ctx := context.WithValue(context.Background(), oauth2.HTTPClient, http.DefaultClient)
 
 	conf, err := google.JWTConfigFromJSON(jsonKey, androidpublisher.AndroidpublisherScope)
 
-	return Client{conf.Client(ctx)}, err
+	return &Client{conf.Client(ctx)}, err
+}
+
+// NewWithClient returns http client which includes the custom http client.
+func NewWithClient(jsonKey []byte, cli *http.Client) (*Client, error) {
+	ctx := context.WithValue(context.Background(), oauth2.HTTPClient, cli)
```

Comment:
oauth2.Transport uses the given client's transport when it is set in the context with ``oauth2.HTTPClient`` key.

https://github.com/golang/oauth2/blob/cdc340f7c179dbbfa4afd43b7614e8fcadde4269/oauth2.go#L318-L323
https://github.com/golang/oauth2/blob/cdc340f7c179dbbfa4afd43b7614e8fcadde4269/internal/transport.go#L25-L29